Update `Open Editors` widget UI

msujew opened the pull request:
#### What it does

Minor changes to focus and hovering effects for the "Open Editors" widget and tree views in general, which were missing their outlines.

#### How to test

1. Open the "Open Editors" view container and add a few files to it
2. The left padding of all nodes should align to vscode
3. The selected node should always display its close icon
4. Hovering over the close icon should show the usual rounded border
5. Hovering over a dirty node should show the close icon.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

colin-grant-work commented:
@msujew, with your tabbar work merged, this should be good to go after a rebase?

msujew commented:
@colin-grant-work Thanks for the reminder, I rebased my changes 👍

colin-grant-work commented on packages/navigator/src/browser/open-editors-widget/navigator-open-editors-widget.tsx:
```diff
@@ -224,7 +225,7 @@ export class OpenEditorsWidget extends FileTreeWidget {
         if (OpenEditorNode.is(node)) {
             const { widget } = node;
-            this.applicationShell.activateWidget(widget.id);
+            this.applicationShell.revealWidget(widget.id);
```
This is still causing a bit of trouble in one case:

1. Open two groups of editors with two editors each.
2. Open 'Open Editors'
3. Select the Open Editors node for the active editor in the currently inactive tabbar.
4. In Theia, nothing happens: the editor is already visible, so the 'reveal' option results in a no-op. In VSCode, the selected editor becomes active (active tabbar colors and e.g. GitLens annotations appear in that editor and disappear from the other editor).

colin-grant-work replied:
Just confirming that this comment still stands after the rebase.

colin-grant-work commented:
@msujew, I believe the scenario I described in [my comment above](https://github.com/eclipse-theia/theia/pull/10940#discussion_r849967341) is still causing trouble. I've removed my self-review request just for bookkeeping purposes - ping me when it's ready for another look.

msujew commented:
@colin-grant-work I had time to get back to this. I removed the functional change and kept it limited to purely visual changes. In my newest change I also corrected the left padding of the tree nodes, which wasn't quite correct.

colin-grant-work reviewed (approved):
Changes and new interactions look good to me. 👍

vince-fugnitto reviewed (commented):
@msujew I noticed a couple of things when quickly trying out the functionality:

1. It seems as though the selection is not maintained similarly to the explorer, and vscode. In the case of the open-editors widget the selection is always passive:

https://user-images.githubusercontent.com/40359487/171458286-ad5b7594-41e7-41c2-959e-a9daf669ccc5.mov

2. I noticed we did not have ellipsis for overflowing content in the open-editors, but it seems to be a regression as it used to work (https://github.com/eclipse-theia/theia/issues/7145). Not related to your changes but I thought I'd mention it here for visibility:

<img width="309" alt="Screen Shot 2022-06-01 at 12 47 37 PM" src="https://user-images.githubusercontent.com/40359487/171458421-97d46bff-e11c-4830-92b9-76a549adf668.png">
